x/distribution/spec/05_hooks.md

```diff
@@ -5,26 +5,53 @@ order: 5
 # Hooks
 
 ## Create or modify delegation distribution
- 
+
  - triggered-by: `staking.MsgDelegate`, `staking.MsgBeginRedelegate`, `staking.MsgUndelegate`
 
-The pool of a new delegator bond will be 0 for the height at which the bond was
-added, or the withdrawal has taken place. This is achieved by setting
-`DelegationDistInfo.WithdrawalHeight` to the height of the triggering transaction. 
+### Before
+
+The delegation rewards are withdrawn to the withdraw address of the delegator.
+The rewards include the current period and exclude the starting period.
+The validator period is incremented.
+The validator period is incremented because the validator's power and share distribution might have changed.
+The reference count for the delegator's starting period is decremented.
+
+### After
+
+The starting height of the delegation is set to the previous period.
+Because of the `Before`-hook, this period is the last period for which the delegator was rewarded.
+
+## Validator created
+
+- triggered-by: `staking.MsgCreateValidator`
+
+When a validator is created, the following validator variables are initialized:
+
+- Historical rewards
+- Current accumulated rewards
+- Accumulated commission
+- Total outstanding rewards
+- Period
+
+By default, all values are set to a `0`, except period, which is set to `1`.
+
+## Validator removed
+
+- triggered-by: `staking.RemoveValidator`
 
-## Commission rate change
- 
- - triggered-by: `staking.MsgEditValidator`
+Outstanding commission is sent to the validator's self-delegation withdrawal address.
+Remaining delegator rewards get sent to the community fee pool.
 
-If a validator changes its commission rate, all commission on fees must be
-simultaneously withdrawn using the transaction `TxWithdrawValidator`.
-Additionally the change and associated height must be recorded in a
-`ValidatorUpdate` state record.
+Note: the validator only gets removed when it has no remaining delegations.
+At that time, all outstanding delegator rewards will have been withdrawn.
+Any remaining rewards are dust amounts.
 
-## Change in Validator State
- 
- - triggered-by: `staking.Slash`, `staking.UpdateValidator`
+## Validator is slashed
 
-Whenever a validator is slashed or enters/leaves the validator group all of the
-validator entitled reward tokens must be simultaneously withdrawn from
-`Global.Pool` and added to `ValidatorDistInfo.Pool`. 
+- triggered-by: `staking.Slash`
+  
+The current validator period reference count is incremented.
+The reference count is incremented because the slash event has created a reference to it.
+The validator period is incremented.
+The slash event is stored for later use.
+The slash event will be referenced when calculating delegator rewards.
```

Comment:
```suggestion
- The current validator period reference count is incremented.
- The reference count is incremented because the slash event has created a reference to it.
- The validator period is incremented.
- The slash event is stored for later use.
- The slash event can be referenced when calculating delegator rewards.
```

Reply:
```suggestion
- The current validator period reference count is incremented.
  The reference count is incremented because the slash event has created a reference to it.
- The validator period is incremented.
- The slash event is stored for later use.
  The slash event will be referenced when calculating delegator rewards.
```